sqlitecluster/SQLite.cpp

```diff
@@ -223,6 +221,88 @@ SQLite::SQLite(const string& filename, int cacheSize, bool enableFullCheckpoints
     sqlite3_progress_handler(_db, 1'000'000, _progressHandlerCallback, this);
 }
 
+SQLite::SQLite(const SQLite& from) :
+    whitelist(nullptr),
+    _sharedData(from._sharedData),
+    _filename(from._filename),
+    _journalSize(from._journalSize),
+    _maxJournalSize(from._maxJournalSize),
+    _insideTransaction(false),
+    _beginElapsed(0),
+    _readElapsed(0),
+    _writeElapsed(0),
+    _prepareElapsed(0),
+    _commitElapsed(0),
+    _rollbackElapsed(0),
+    _mutexLocked(false),
+    _enableRewrite(false),
+    _currentlyRunningRewritten(false),
+    _timeoutLimit(0),
+    _abandonForCheckpoint(false),
+    _autoRolledBack(false),
+    _noopUpdateMode(false),
+    _enableFullCheckpoints(from._enableFullCheckpoints),
+    _queryCount(0),
+    _cacheHits(0),
+    _useCache(false),
+    _isDeterministicQuery(false),
+    _pageLoggingEnabled(from._pageLoggingEnabled),
+    _currentTransactionAttemptCount(-1),
+    _cacheSize(from._cacheSize),
+    _synchronous(from._synchronous),
+    _mmapSizeGB(from._mmapSizeGB)
+{
+    SINFO("Opening sqlite database: " << _filename);
```

Comment:
nab, kinda sucks we duplicated all of this code. not sure i have a real proposal for how to resolve that. maybe a function?

Reply:
We can probably do this with delegating constructors: https://docs.microsoft.com/en-us/cpp/cpp/delegating-constructors?view=vs-2019

I don't care enough to change it right now when everything is working, though.